### ISSUE TITLE
Driver Installation Failed on Big Sur

### ISSUE THREAD

**Issue by djhalnon** [360Controller/360Controller#1148]
### Type of Controller
N/A

### OS Version
Mac OS Big Sur 11.2.2

### Driver Version
1.0.0-alpha.6

### Connection Method
N/A

### Device Name and Info
N/A

### Response of Controller
N/A

### Response in Preference Pane
N/A

### Detailed Explaination of Issue
Installation of driver fails. The error message is "The installation failed. The installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance." This error message does not disclose the actual nature of the error that caused the installer to fail.

<img width="626" alt="InstallError" src="https://user-images.githubusercontent.com/19252010/109399221-881cef00-790f-11eb-9e1b-14385cb53394.png">

**Comment by Zara4397** [comment]
Got the same result, but got also a pop-up from Mac stating there are instances that needed to be updated. Probably just a matter of supporting Big Sur.

**Comment by paulina222** [comment]
> Got the same result, but got also a pop-up from Mac stating there are instances that needed to be updated. Probably just a matter of supporting Big Sur.

the same thing happened to me.. Zara4397, did you ever find another way to use your xbox controller on your mac by any chance??

**Comment by JamesDeCarlo** [comment]
Same Issue with Big Sur Version 11.2.2

**Comment by romulobastos** [comment]
Same issue with Big Sur V 11.2.3

**Comment by alfanse** [comment]
Same failure, MacOs v11.2.3
Existing install was failing to detect device, turning on Driver failed due to inability to add "Drew <something>" to permitted list of app developers.
Attempted re-install and was presented with error described in posting above, only thing i can add is the console log from the failed install:
```
2021-03-22 17:47:50+00 my-MacBook-Pro package_script_service[1709]: PackageKit: Executing script "postinstall" in /tmp/PKInstallSandbox.zCtfsq/Scripts/com.mice.pkg.Xbox360Controller.VtNyyT
2021-03-22 17:47:50+00 my-MacBook-Pro package_script_service[1709]: Set responsibility to pid: 1825, responsible_path: /System/Library/CoreServices/Installer.app/Contents/MacOS/Installer

<hmmmm>
2021-03-22 17:47:51+00 my-MacBook-Pro package_script_service[1709]: ./postinstall: Major version is not 10. Cowardly failing out of script execution.
</hmmmm>

2021-03-22 17:47:51+00 my-MacBook-Pro package_script_service[1709]: Responsibility set back to self.
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: releasing Spotlight indexing
2021-03-22 17:47:51+00 my-MacBook-Pro install_monitor[1828]: Re-included: /Applications, /Library, /System, /bin, /private, /sbin, /usr
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: releasing backupd
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: allow user idle system sleep
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: Cleared responsibility for install from 1825.
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: Cleared permissions on Installer.app
2021-03-22 17:47:51+00 my-MacBook-Pro installd[1242]: PackageKit: Install Failed: Error Domain=PKInstallErrorDomain Code=112 "An error occurred while running scripts from the package “Install360Controller.pkg”." UserInfo={NSFilePath=./postinstall, NSURL=file:///Volumes/360ControllerInstall/Install360Controller.pkg#Install360Controller.pkg, PKInstallPackageIdentifier=com.mice.pkg.Xbox360Controller, NSLocalizedDescription=An error occurred while running scripts from the package “Install360Controller.pkg”.} {
	    NSFilePath = "./postinstall";
	    NSLocalizedDescription = "An error occurred while running scripts from the package \U201cInstall360Controller.pkg\U201d.";
	    NSURL = "file:///Volumes/360ControllerInstall/Install360Controller.pkg#Install360Controller.pkg";
	    PKInstallPackageIdentifier = "com.mice.pkg.Xbox360Controller";
	}
```
Hope that helps?

**Comment by L0VEFIRE** [comment]
its going to fail it wasn't made for to work Big Sur the app is dead on Big Sur, you have to manually approve the driver but that is kinda pointless now since it won't work even after you do.
you need to start using bluetooth controller supported by apple. Xbox one or newer / PS4 or newer. unless you go back to Catalina and use this driver. 

if you using steam, then you need to load big picture modem and download the gamepad driver there. anywhere else should just work wirelessly.

**Comment by djhalnon** [comment]
Interesting. What's behind the decision to prevent people from using wired controllers on Macs? It's one thing not to sell them, but another to actively prevent people who already have them, from ever using one. I honestly don't get it.

**Comment by JamesDeCarlo** [comment]
Definitely disappointing. Used this for years on my mac before upgrading to big sur. I have newer xbox one controllers with bluetooth that work. But this driver also worked excellent for my logitech 710 controller. Which I miss. Thank you for the response though.

**Comment by L0VEFIRE** [comment]
> Interesting. What's behind the decision to prevent people from using wired controllers on Macs? It's one thing not to sell them, but another to actively prevent people who already have them, from ever using one. I honestly don't get it.


Honestly, i asked apple many times in tge last year with no response, i even send product feedback, i think more ppl need to do in order to let them know how important it is to ppl. though it might just be better if everyone just mailed tim cook gets more attention tcook@apple.com 
anyway it is very disappointing i agree, im forced to use boot camp to play games now.

**Comment by sashatc** [comment]
Got my xbox one controller to work wired with macOS 11.2.3. Earlier I was struggling to "allow" driver from Drew Mills because
I there was simply no prompt in Security settings to Allow it. 

Inspired by https://steamcommunity.com/discussions/forum/2/3004429475624592660/  I went and figured out signers Team ID: G947KKM5RL

so, booted into Recovery Mode and executed:
`spctl kext-consent add G947KKM5RL`

followed by:
`kmutil trigger-panic-medic --volume-root "/Volumes/**{{replace with your macOS drive name}}**"`

upon reboot in Panic Medic mode I've been asked to allow all the third-party extensions by developers. Drew Mills was not one of them. After allowing those macOS prompted to reboot but I chose "Not Now" option and instead went to XBOX 360 Controller Pref panel - Advanced Tab and clicked "Enable Driver" (it was unchecked), wait second or two and get notification that I need to approve software from Drew Mills in Security settings - went to Security Settings and hurray - this time there was a prompt to do so! Allowed, rebooted and it works!
<img width="940" alt="Screen Shot 2021-04-05 at 5 08 37 PM" src="https://user-images.githubusercontent.com/12070484/113641545-b094b900-9632-11eb-95d9-35a2bf756e17.png">
<img width="855" alt="Screen Shot 2021-04-05 at 5 09 05 PM" src="https://user-images.githubusercontent.com/12070484/113641556-b5596d00-9632-11eb-94f6-ed656d0f2307.png">

**Comment by L0VEFIRE** [comment]
@iSashaCH 
yeah nice, not sure how you got your controller working. i already have a big post that explains this here. sorry you had to do it the longer way. still curious what made yours work though.

**Comment by diego-lipinski-de-castro** [comment]
I followed @iSashaCH instructions and also got it to work on Big Sur

<img width="1648" alt="Captura de Tela 2021-04-23 às 20 03 46" src="https://user-images.githubusercontent.com/29608698/115937765-03fa6a00-a46f-11eb-9e17-bf48fe470072.png">

**Comment by L0VEFIRE** [comment]
I'm glad you got it to work but I have a whole post on this that was made prior to this, why are we creating a new duplicate thread. 🤦 

Link:
https://github.com/360Controller/360Controller/issues/1139

**Comment by alexochs** [comment]
@iSashaCH's approach seems to **not** work on M1 Macs :(

**Comment by L0VEFIRE** [comment]
> @iSashaCH's approach seems to **not** work on M1 Macs :(

As of currently, you need to use Windows on a M1 mac to get controllers to work. (This can be done via VM Software like Parrells or VMware … ect) 
I’m finally ordering a M1 mac tomm so when i get it i can do some more investigation work.

**Comment by ruyrybeyro** [comment]
Following these steps,   the last version of the extension  appears grayed out in the Security setting of Big Sure. It let´s itsef be selected, but does not appear loaded at boot. Any comments?

**Comment by letalvoj** [comment]
Solution which worked for me: https://github.com/360Controller/360Controller/issues/1180#issuecomment-986002973

First install `0.16.11`, make sure you get to the point where the settings appear in the system, yet it says no device. Then install `1.0.0-alpha6`.

**Comment by alvarusan** [comment]
Solution that worked for me on 12.6.8 (Monterey) [#1148 comment](https://github.com/360Controller/360Controller/issues/1148#issuecomment-813735388)

**Comment by Darkdonald2** [comment]
Disabling system integrity protection (SIP) then trying to enable the driver again worked for me on macOS Sonoma.
A search on google will tell you how to disable SIP  
Unfortunately re-enabling SIP disables the driver. 
hope that helps
